### DISCUSSION TITLE
vcpkg makes my binary larger even though it doesn't use any ports, how can I see what is linked?

### DISCUSSION THREAD

**Discussion by jdx-john** [microsoft/vcpkg#20383] [Q&A]
I recently found one of our modules suffers a regression - WinAPI functions fail - when vcpkg is enabled (which it was by default). It took a while to identify this as the cause but once I did I also noticed that when I disable vcpkg in that VC++ project, the DLL shrank by about 40KB.

It's all rather perplexing, what is vcpkg doing and how can that impact my run-time performance? My _guess_ is maybe some file is doing something odd to a `#define ...` because Windows APIs can be quite fragile.

But my question is, can I get a list of _exactly_ what ports/includes/libs vcpkg is pulling in automagically when I build a project?

**Comment by autoantwort** [comment]
msbuild? then all libs are pulled in. The vcpkg include folder is added to the include path.

**Comment by jdx-john** [comment]
> msbuild? then all libs are pulled in. The vcpkg include folder is added to the include path.

I guess I mean, can I see what is actually linked then... I have hundreds of MB of libs but only 40KB larger on this particular module. I wonder what the compiler/linker is doing differently.

**Comment by autoantwort** [comment]
Hm I have no idea. You I think you could run `DUMPBIN /SYMBOLS` on your executable and all libs and then see which symbols are from where. Probably you have to write a script for that by your amount of libs.

**Comment by jdx-john** [comment]
That's a pain. I'd rather hoped there might be an equivalent of "show includes" which lists every include file pulled in. It's a perplexing bug that a WinAPI function fails when vcpkg is enabled, considering the project builds without it. The linker wouldn't normally link anything not being used so somewhat a mystery :)
